### PR TITLE
REL-983 v23.1.23: [Docs] Generate release notes for Cloud

### PR DESCRIPTION
**Pull request by florence-crl**
Fixes REL-983

(1) edited notes, (2) added cloud fields to releases.yml

Rendered preview: 
- [releases list page](https://deploy-preview-18663--cockroachdb-docs.netlify.app/docs/releases/#v23-1)
- [v23.1.23](https://deploy-preview-18663--cockroachdb-docs.netlify.app/docs/releases/v23.1.html#v23-1-23)

### CODE REVIEW

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+<h3 id="v23-1-23-sql-language-changes">SQL language changes</h3>
+
+- Introduced three new [cluster settings]({% link v23.1/cluster-settings.md %}) for controlling table statistics forecasting:
+    - [`sql.stats.forecasts.min_observations`]({% link v23.1/cluster-settings.md %}) is the minimum number of observed statistics required to produce a forecast.
+    - [`sql.stats.forecasts.min_goodness_of_fit`]({% link v23.1/cluster-settings.md %}) is the minimum R² (goodness of fit) measurement required from all predictive models to use a forecast.
+    - [`sql.stats.forecasts.max_decrease`]({% link v23.1/cluster-settings.md %}) is the most a prediction can decrease, expressed as the minimum ratio of the prediction to the lowest prior observation. [#124076][#124076]
+- Added a new [session setting]({% link v23.1/session-variables.md %}) `optimizer_use_improved_multi_column_selectivity_estimate`, which if enabled, causes the [optimizer]({% link v23.1/cost-based-optimizer.md %}) to use an improved selectivity estimate for multi-column predicates. This setting will default to `true` on v24.2 and later, and `false` on earlier versions. [#124076][#124076]
```
Comment (taroface) [inline src/current/_includes/releases/v23.1/v23.1.23.md:22]: Is this really referencing v24.2? Just double-checking that we're ok calling out a version that isn't even released yet.

File: src/current/_data/releases.yml [hunk 2]
```diff
@@ -6178,3 +6178,37 @@
...
+    docker_arm_limited_access: false
+  source: true
+  previous_release: v23.1.22
+  cloud_only: True
+  cloud_only_message_short: 'Available only for select CockroachDB Cloud clusters'
+  cloud_only_message: >
+      This version is currently available only for select
+      CockroachDB Cloud clusters. To request to upgrade
+      a CockroachDB self-hosted cluster to this version,
+      [contact support](https://support.cockroachlabs.com/hc/en-us/requests/new).
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_data/releases.yml]: good catch! updated

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+## v23.1.23
+
+Release Date: June 20, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-1-23-{{-site.data.products.enterprise-}}-edition-changes">{{ site.data.products.enterprise }} edition changes</h3>
+
+- Fixed a bug that was present since v22.2 where [changefeeds]({% link v23.1/change-data-capture-overview.md %}) with long-running [initial scans]({% link v23.1/create-changefeed.md %}#initial-scan) might incorrectly restore checkpoint job progress and drop events during [changefeed restarts]({% link v23.1/changefeed-messages.md %}#duplicate-messages) due to transient errors or node restarts. The bug was most likely to occur in clusters with the following contributing factors:
+    - The cluster settings `changefeed.frontier_checkpoint_frequency` and `changefeed.frontier_highwater_lag_checkpoint_threshold` were set low, which resulted in the initial scan taking many multiples of the configured frequency to complete.
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated, added link

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    - There were multiple target tables with significant differences in row counts in one changefeed.
+    - The changefeed target tables were large with many ranges.
+    - The initial scan took a long time to complete (an hour or longer). [#123970][#123970] [#124759][#124759]
+- [Changefeeds]({% link v23.1/change-data-capture-overview.md %}) now default to evenly distributing their work across all replicas, including followers, regardless of leaseholder placement. To disable this behavior, set the [cluster setting]({% link v23.1/cluster-settings.md %}) `changefeed.balanced_distribution.enabled` to `false`. If disabled, changefeed planning reverts to its previous behavior for distributing work. [#124930][#124930]
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated, added links

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+<h3 id="v23-1-23-sql-language-changes">SQL language changes</h3>
+
+- Introduced three new [cluster settings]({% link v23.1/cluster-settings.md %}) for controlling table statistics forecasting:
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated, added link

File: src/current/_includes/releases/v23.1/v23.1.23.md (hunk 1 above)
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md:22]: I copied this note from [v24.1.0-rc.1](https://www.cockroachlabs.com/docs/releases/v24.1.html#v24-1-0-rc-1-sql-language-changes) and [v23.2.6](https://www.cockroachlabs.com/docs/releases/v23.2.html#v23-2-6-sql-language-changes).

So I will keep the reference to v24.2.

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    - [`sql.stats.forecasts.min_observations`]({% link v23.1/cluster-settings.md %}) is the minimum number of observed statistics required to produce a forecast.
+    - [`sql.stats.forecasts.min_goodness_of_fit`]({% link v23.1/cluster-settings.md %}) is the minimum R² (goodness of fit) measurement required from all predictive models to use a forecast.
+    - [`sql.stats.forecasts.max_decrease`]({% link v23.1/cluster-settings.md %}) is the most a prediction can decrease, expressed as the minimum ratio of the prediction to the lowest prior observation. [#124076][#124076]
+- Added a new [session setting]({% link v23.1/session-variables.md %}) `optimizer_use_improved_multi_column_selectivity_estimate`, which if enabled, causes the [optimizer]({% link v23.1/cost-based-optimizer.md %}) to use an improved selectivity estimate for multi-column predicates. This setting will default to `true` on v24.2 and later, and `false` on earlier versions. [#124076][#124076]
+- The row-level TTL setting [`ttl_delete_rate_limit`]({% link v23.1/row-level-ttl.md %}) is now set to `100` by default, which sets the rate limit for deleting expired rows to `100`. [#124362][#124362]
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated, added link

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+- CockroachDB no longer limits precision when converting [spatial data types]({% link v23.1/architecture/glossary.md %}#data-types) to JSON. [#124534][#124534]
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated, added link

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+<h3 id="v23-1-23-operational-changes">Operational changes</h3>
+
+- - The `client_authentication_ok` and `client_session_end` events are now logged to the [`SESSIONS` log channel]({% link v23.1/logging-use-cases.md %}#sessions) unconditionally. Previously, these would only be logged if the `server.auth_log.sql_sessions.enabled` [cluster setting]({% link v23.1/cluster-settings.md %}) was set to `true`. All other `SESSIONS` log messages are still only logged if `server.auth_log.sql_sessions.enabled` or `server.auth_log.sql_connections.enabled` are set to `true`. To not show `client_authentication_ok` and `client_session_end` events, disable the `SESSIONS` log channel entirely. [#124375][#124375]
+
+<h3 id="v23-1-23-db-console-changes">DB Console changes</h3>
+
+- Viewing the [**SQL Activity**]({% link v23.1/ui-statements-page.md %}#statement-fingerprints-results) sorted by `% of Runtime` now correctly sorts entries by the runtime amount. [#123899][#123899]
+- For the [DB Console]({% link v23.1/ui-overview.md %}), the favicon and other image files now render properly. [#122702][#122702]
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated

File: src/current/_includes/releases/v23.1/v23.1.23.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+<h3 id="v23-1-23-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug where a failed [restore]({% link v23.1/restore.md %}) job could leave the system in a state where re-attempting the restore was not possible without manual intervention. [#123462][#123462]
+- [Index recommendations]({% link v23.1/ui-databases-page.md %}#index-recommendations) in the [DB Console]({% link v23.1/ui-overview.md %}) will now function properly for indexes on tables or columns whose names contain quotation marks or whitespace. For example: `CREATE INDEX ON "my table" ("my col");`. [#122117][#122117]
+- Fixed a bug where [client certificate authentication]({% link v23.1/authentication.md %}#client-authentication) combined with [identity maps]({% link v23.1/sso-sql.md %}#identity-map-configuration) (`server.identity_map.configuration`) did not work since v23.1. For the feature to work correctly, the client must specify a valid db user in the [connection string]({% link v23.1/connection-parameters.md %}). [#124076][#124076]
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/_includes/releases/v23.1/v23.1.23.md]: updated

**Review by florence-crl — COMMENTED** [review]
TFTR!

**Pull request merged**